lib/rules/no-magic-numbers.js

```diff
@@ -90,6 +95,18 @@ module.exports = {
             return ignore.indexOf(value) !== -1;
         }
 
+        /**
+         * Returns whether the number should be ignore when used as a default
+         * value assignment.
+         * @param {ASTNode} fullNumberNode `Literal` or `UnaryExpression` full number node
+         * @returns {boolean} true if the number should be ignored
+         */
+        function isIgnoredDefaultValue(fullNumberNode) {
+            const parent = fullNumberNode.parent;
+
+            return parent.type === "AssignmentPattern" && ignoreDefaultValues;
```

Comment:
Can we also add `parent.right === fullNumberNode` condition, e.g.:

```js
return parent.type === "AssignmentPattern" && parent.right === fullNumberNode && ignoreDefaultValues;
```

Reply:
Done.